src/datepicker.jsx

```diff
@@ -188,7 +188,7 @@ var DatePicker = React.createClass({
       return
     }
 
-    if (!isSameDay(this.props.selected, changedDate)) {
+    if (!isSameDay(this.props.selected, changedDate) || this.props.disableDateAutoCorrection) {
```

Comment:
If I understand correctly what is happening does this not cause the `onChange` event to fire each times... For example if the `disableDateAutoCorrection` flag set to true and there is a valid date in the field if the user enters the field and then leaves without changing the data the `onChange` will fire because of the  `disableDateAutoCorrection` flag. All that said I could be wrong I am the "New Guy"™